If response is json embedded in soap, then error should not be thrown. Fix issue #580

Himanshu4 opened the pull request:
Changes included:
1. Changed console.log to debug when soap response is json to avoid unnecessary logs.
2. Fix for issue #580 . When json is embedded in soap body, then currently it throws error. We won't return the error and send the response in body param. That way, it would be a cleaner approach for the client to distinguish it from other errors like "http error"/"soap error".

herom commented:
Thanks a lot for your contribution @Himanshu4 - and sorry for the long delay :disappointed:. I'm getting the feeling of being the only contributor who cares for this project at the moment :smiley:

Personally, I'm totally happy with the addition of the `debug` library and I hope we make use of it more often so that this library can "speak about it's work" to the developer which would help a lot! So, whoever reads this and want to get a hand on pulling in `debug`  wherever necessary/helpful, this would make a great PR :+1: 

In order to get your PR merged in, I also have to ask you for accompanying tests, so we can make shure that your code changes/additions don't get overwritten accidently. Please have a look at our [Guidelines on Submitting a Pull Request](https://github.com/vpulim/node-soap/blob/master/CONTRIBUTING.md#submitting-a-pull-request) in advance :smiley:

Himanshu4 commented:
@herom : Apologies for so long delay. I have added a test wsdl in which response element is not defined as it is a json and also added a test case for the same. Combined both the previous commits into a single one. Please check and let me know if it can be merged now.

Thanks.

herom commented:
Awesome, thanks a lot :+1:

Himanshu4 commented:
@herom : In which version, would this change be start reflecting in the npm module.

herom commented:
@Himanshu4 I'm planning on releasing a new version of `node-soap` hopefully by Wednesday/Thursday this week :+1:
